### PR TITLE
DP-17958 [React] Add button disabled states and fix highlight theme a11y

### PR DESCRIPTION
**Pull request by l0c0luke**
## Description

* Disables `:hover` rules for `:disabled` buttons in all themes and usages.
* Adjustments to the `c-highlight` theme to resolve accessibility errors.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-17958)
- [Github issue](https://github.com/massgov/mayflower/issues/884)

## Steps to Test

1. Goto http://localhost:6006/?path=/story/atoms-buttons--button and follow the instructions in the GitHub Issue created by @clairesunstudio  

#### Impacted Areas in Application

* Buttons!

### CODE REVIEW

File: changelogs/DP-17958.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Every PR must have a changelog. To create a changelog:
```
Comment (clairesunstudio) [inline changelogs/DP-17958.yml]: Can you remove all the commented out code in the changelog

**Review by clairesunstudio — COMMENTED** [review]
Changes look good!

**Comment by clairesunstudio** [comment]
Also, since there are no themes for tertiary and quaternary usages, can you in storybook conditionally render themes field for those usages (or make the option array empty)

**Comment by l0c0luke** [comment]
> Also, since there are no themes for tertiary and quaternary usages, can you in storybook conditionally render themes field for those usages (or make the option array empty)

I couldn't find a way to do this. Seems the selects are not updated after initially being drawn and I could not find any API to access the knobs after the fact. Not sure it's worth spending too much more time on this as I believe they are adding this sort of functionality to `v6` of storybook.

**Review by clairesunstudio — APPROVED** [review]

**Pull request merged**